opentelemetry-api/src/opentelemetry/trace/status.py

```diff
@@ -0,0 +1,168 @@
+    FAILED_PRECONDITION, ABORTED, and UNAVAILABLE: (a) Use UNAVAILABLE if the
+    client can retry just the failing call. (b) Use ABORTED if the client should
+    retry at a higher level (e.g., when a client-specified test-and-set fails,
+    indicating the client should restart a read-modify-write sequence). (c)
+    Use FAILED_PRECONDITION if the client should not retry until the system state
+    has been explicitly fixed. E.g., if an "rmdir" fails because the directory is
+    non-empty, FAILED_PRECONDITION should be returned since the client should not
+    retry unless the files are deleted from the directory.
+    """
+    FAILED_PRECONDITION = 9
+
+    """The operation was aborted, typically due to a concurrency issue such as a
+    sequencer check failure or transaction abort. See the guidelines above for
+    deciding between FAILED_PRECONDITION, ABORTED, and UNAVAILABLE."""
+    ABORTED = 10
+
+    """The operation was attempted past the valid range. E.g., seeking or reading
+    past end-of-file. Unlike INVALID_ARGUMENT, this error indicates a problem that
+    may be fixed if the system state changes. For example, a 32-bit file system
+    will generate INVALID_ARGUMENT if asked to read at an offset that is not in
+    the range [0,2^32-1],but it will generate OUT_OF_RANGE if asked to read from
+    an offset past the current file size. There is a fair bit of overlap between
+    FAILED_PRECONDITION and OUT_OF_RANGE. We recommend using OUT_OF_RANGE
+    (the more specific error) when it applies so that callers who are iterating
+    through a space can easily look for an OUT_OF_RANGE error to detect when they
+    are done.
+    """
+    OUT_OF_RANGE = 11
+
+    """The operation is not implemented or is not supported/enabled in this
+    service."""
+    UNIMPLEMENTED = 12
+
+    """Internal errors. This means that some invariants expected by the
+    underlying system have been broken. This error code is reserved for
+    serious errors.
+    """
+    INTERNAL = 13
+
+    """The service is currently unavailable. This is most likely a transient
+    condition, which can be corrected by retrying with a backoff.
+    Note that it is not always safe to retry non-idempotent operations.
+    """
+    UNAVAILABLE = 14
+
+    """Unrecoverable data loss or corruption."""
+    DATA_LOSS = 15
+
+    """The request does not have valid authentication credentials for the
+    operation.
+    """
+    UNAUTHENTICATED = 16
+
+
+class Status:
+    """Represents the status of a finished Span
+
+    Args:
+        canonical_code: Represents the canonical set of status codes of a
+        finished Span
+        description: Description of the Status.
+    """
+
+    def __init__(
+        self,
+        canonical_code: "StatusCanonicalCode" = StatusCanonicalCode.OK,
+        description: typing.Optional[str] = None,
+    ):
+        self.code = canonical_code
+        self.desc = description
+
+    @property
```

Comment:
Why make these properties instead of just exposing the attributes?

Reply:
@Oberon00  commented before related this, using properties would be more pythonic, I'm fine with any approach 

https://github.com/open-telemetry/opentelemetry-python/pull/213#discussion_r333360636

Reply:
I agree they shouldn't be getters too, I just don't see the point of exposing existing attrs under a different name.

I think it'd be fine to just expose `code` and `desc`, but if you want to hide them behind attributes I think you should change the underlying attrs to `_code` and `_desc` (or more conventionally: `_canonical_code`, `_description`).

Reply:
@hectorhdzg: Could you please comment on why you made them properties instead of simply public attributes?

Reply:
is_ok require a special getter because is doing an operation to get the value, I guess other two could be just simple attributes. I felt is just cleaner to have them as properties as well, I can update if there are strong opinions about this

Reply:
No strong opinion as far as I'm concerned, it's just surprising to see properties wrap public attributes like this. We have both properties and bare attributes in the API now, and don't really have consistent reasons for using one over the other.